src/savedata.c

```diff
@@ -845,585 +838,558 @@ int SaveTableEntry_BodySize (int saveTableID)
+            saveData->sectorSwitch = SECTOR_ID_BACKUP;
+            saveData->sectorCounter = backupCounter;
+            SaveData_CardLoad((BACKUP_SECTOR_START + saveTable->blockID) * SAVE_SECTOR_SIZE, ret, size);
+            return ret;
         }
     }
 
-    *param3 = 2;
+    *loadResult = LOAD_RESULT_CORRUPT;
 
-    param0->unk_202C4 = 0;
-    param0->unk_202C8 = 0;
+    saveData->sectorSwitch = 0;
+    saveData->sectorCounter = 0;
 
-    return v1;
+    return ret;
 }
 
-void * SaveDataExtra_Mirror (SaveData * param0, int param1, int param2, int * param3, BOOL * param4)
+void* SaveDataExtra_Mirror (SaveData *saveData, int heapID, int extraSaveID, int *loadResult, BOOL *isOld)
 {
-    const SaveTableEntry * v0;
-    void * v1;
-    u32 v2;
-    BOOL v3, v4;
-    u32 v5, v6;
-    u32 v7, v8;
-    u8 v9;
-    UnkStruct_0202783C * v10 = sub_0202783C(param0);
+    const SaveTableEntry *saveTable;
+    void *ret;
+    u32 size;
+    BOOL primaryResult, backupResult;
+    u32 primaryKey, backupKey, currKey, oldKey;
+    u8 keyFlag;
+    UnkStruct_0202783C *miscSave = sub_0202783C(saveData);
 
-    GF_ASSERT(param2 < gExtraSaveTableSize);
-    GF_ASSERT(param2 != 0);
+    GF_ASSERT(extraSaveID < gExtraSaveTableSize);
+    GF_ASSERT(extraSaveID != EXTRA_SAVE_TABLE_ENTRY_HALL_OF_FAME);
 
-    v0 = &gExtraSaveTable[param2];
-    GF_ASSERT(v0->dataID == param2);
+    saveTable = &gExtraSaveTable[extraSaveID];
+    GF_ASSERT(saveTable->dataID == extraSaveID);
 
-    v2 = v0->sizeFunc() + sizeof(UnkStruct_020253B4);
-    v1 = Heap_AllocFromHeap(param1, v2);
+    size = saveTable->sizeFunc() + sizeof(SaveCheckFooter);
+    ret = Heap_AllocFromHeap(heapID, size);
 
-    sub_020259F8(param0, param2, &v7, &v8, &v9);
-    sub_02025AC0((0 + v0->blockID) * 0x1000, v1, v2);
+    SaveDataExtra_SaveKey(saveData, extraSaveID, &currKey, &oldKey, &keyFlag);
+    SaveData_CardLoad((PRIMARY_SECTOR_START + saveTable->blockID) * SAVE_SECTOR_SIZE, ret, size);
 
-    v3 = sub_020253E0(param0, v1, param2, v0->sizeFunc());
-    MI_CpuCopy8(v1, &v5, sizeof(u32));
+    primaryResult = SaveCheckFooter_Validate(saveData, ret, extraSaveID, saveTable->sizeFunc());
+    MI_CpuCopy8(ret, &primaryKey, sizeof(u32));
 
-    sub_02025AC0((64 + v0->blockID) * 0x1000, v1, v2);
-    v4 = sub_020253E0(param0, v1, param2, v0->sizeFunc());
+    SaveData_CardLoad((BACKUP_SECTOR_START + saveTable->blockID) * SAVE_SECTOR_SIZE, ret, size);
+    backupResult = SaveCheckFooter_Validate(saveData, ret, extraSaveID, saveTable->sizeFunc());
 
-    MI_CpuCopy8(v1, &v6, sizeof(u32));
+    MI_CpuCopy8(ret, &backupKey, sizeof(u32));
 
-    *param3 = 1;
-    *param4 = 0;
+    *loadResult = LOAD_RESULT_OK;
+    *isOld = FALSE;
 
-    if ((v3 == 1) && (v4 == 0)) {
-        if (v7 == v5) {
-            if (v9 == 1) {
-                sub_02025A18(param0, param2, v8, v8, 0);
-                *param4 = 1;
+    if (primaryResult == TRUE && backupResult == FALSE) {
+        if (currKey == primaryKey) {
```

Comment:
I think it reads better if this second condition is folded into the first (and it reduces block-indentation).

```suggestion
    if (primaryResult == TRUE
            && backupResult == FALSE
            && currKey == primaryKey) {
```